### PR TITLE
#168781633 - Manager Approve Reject Requests

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Manager Approve Reject Requests.

#### Description of Task to be completed?
- Create processReqquestAction.
- Create ProcessRequest reusable button component.
- Create tests.

#### How should this be manually tested?
- Clone ft-approve-reject-requests-168781633 branch
- run `npm i`
- run `npm run dev`
- Login as a Manager and click on `Manage Requests` in the Menu

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781633](https://www.pivotaltracker.com/story/show/168781633)

#### Screenshots (if appropriate)
Screen Shot 2019-12-31 at 17 25 58
Screen Shot 2019-12-31 at 17 45 35

#### Questions:
N/A

### CODE REVIEW

File: src/components/pages/requests/ProcessRequest.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Button } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { processRequestAction, singleRequestAction } from '../../../actions/requestsActions';
+import { getManagerRequestAction } from '../../../actions/managerRequestAction';
+
+
```
Comment (ngireric123) [inline src/components/pages/requests/ProcessRequest.js]: remove one line

File: src/styles/requests/requests.scss [hunk 2]
```diff
@@ -31,4 +31,7 @@
 }
 .reset-cross:hover {
   cursor: pointer;
+}
+.process-request-button {
+  text-transform: uppercase;
 }
```
Comment (ngireric123) [inline src/styles/requests/requests.scss]: add new line

File: src/tests/reducersTests/processRequestReducerTest.spec.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    const returnedSate = processRequestReducer(undefined, action);
+    expect(returnedSate).toEqual({
+      "processData": null,
+      "processError": action.payload,
+      "status": 'error',
+    });
+  });
+
+});
```
Comment (ngireric123) [inline src/tests/reducersTests/processRequestReducerTest.spec.js]: add new line

**Review by ngireric123 — CHANGES_REQUESTED** [review]

**Review by ngireric123 — APPROVED** [review]
LGTM

**Review by sabin18 — CHANGES_REQUESTED** [review]
Good job @nignanthomas , put approve and reject button below the view request button to make it look good.

File: src/components/pages/requests/ProcessRequest.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { getManagerRequestAction } from '../../../actions/managerRequestAction';
+
+export class ProcessRequest extends Component {
+  state = {
+    isLoading: false,
+  };
+
+  processAction = async () => {
+    const { props } = this;
+    const { action, id } = props;
```
Comment (nakiwuge) [inline src/components/pages/requests/ProcessRequest.js]: Add a new line

File: src/components/pages/requests/ProcessRequest.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    this.setState({ isLoading: true });
+    await props.processRequestAction(action, id);
+    await props.getManagerRequestAction();
+    await props.singleRequestAction(id);
+    this.setState({ isLoading: false });
+  };
+
+  render() {
+    const { isLoading } = this.state;
+    const { action, variant } = this.props;
```
Comment (nakiwuge) [inline src/components/pages/requests/ProcessRequest.js]: Add a new line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @nignanthomas. Kindly work on the feedback below.
- The user should receive a confirmation message when the click `Approve/Reject`. Something like `Are you sure you want to approve/reject this request?`. This should be a reusable component because it will also be used on DELETE.
- Disable the `Approve` button when the loader is on the `Reject` button and vice versa
- Go through other comments I have left on the pr

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**